main.go

```diff
@@ -270,6 +288,8 @@ func main() {
 							metricsv1.Logger(logger),
 							metricsv1.Registry(reg),
 							metricsv1.HandlerInstrumenter(ins),
+							metricsv1.ReadMiddleware(authorization.WithAuthorizer(authorizer, rbac.Read, "metrics")),
+							metricsv1.WriteMiddleware(authorization.WithAuthorizer(authorizer, rbac.Write, "metrics")),
```

Comment:
While this feels a bit weird, and I would prefer to be explicit, it allows us to inject different middlewares from the outside to the package, without the need to change anything in the package, when changing middlewares. 
I guess that's the reasoning?

Reply:
yes exactly that's the idea. I want to keep the same package API, where the `New...` func simply returns a `http.Handler` and the package has no concept of the authorization/authentication middlewares. I think the only nice way to do that is to inject middlewares into the creator func. The other option is to make the API packages accept an `rbac.Authorizer` option and build and use the middlewares themselves. WDYT?

Reply:
I think it's fine for now. We might need to come back to the version you just said, but that might be ahead of time right now.

Reply:
:+1: